lib/minefield.rb

```diff
@@ -0,0 +1,198 @@
+require_relative 'cell'
+
+class Minefield
+
+  attr_reader :size, :field
+
+  def initialize(size = 10)
+    set_field_size(size)
+    create_field_with_attributes
+  end
+
+  def render
+    field.each do |row|
+      row.each { |cell| print cell }
+      print "\n"
+    end
+    nil
+  end
+
+  def take_turn(turn)
+    row = turn[:row]
+    column = turn[:column]
+    action = turn[:action]
+    take_action(row, column, action)
+  end
+
+  # GAME ENDING SCENARIOS
+
+  def lost?
+    field.flatten.any? { |cell| cell.exploded? }
+  end
+
+  def blow_up_board
+    field.flatten.each do |cell|
+      cell.clear if cell.mine
+    end
+  end
+
+  def won?
+    uncleared_cells == number_of_mines
+  end
+
+  def flag_remaining_mines
+    field.flatten.each do |cell|
+      cell.flag if cell.mine
+    end
+  end
+
+  def game_over?
+    lost? || won?
+  end
+
+  # PUBLIC BOARD INFORMATION
+
+  def number_of_mines
+    field.flatten.count { |cell| cell.mine }
+  end
+
+  def number_of_flags
+    field.flatten.count { |cell| cell.flagged }
+  end
+
+  def unflagged_mines
+    number_of_mines - number_of_flags
+  end
+
+  def uncleared_cells
+    field.flatten.count { |cell| !(cell.cleared) }
+  end
+
+  private
+
+  # INITIALIZE HELPERS
+
+  def set_field_size(size)
+    if size.class == Fixnum
+      @size = size
```

Comment:
Group instance variables together! it makes it easier to read through code (or use comments)